scipy/stats/_continuous_distns.py

```diff
@@ -3431,29 +3431,57 @@ def g(n):
         g3 = g(3)
         g4 = g(4)
         g2mg12 = np.where(abs(c) < 1e-7, (c*np.pi)**2.0/6.0, g2-g1**2.0)
-        gam2k = np.where(abs(c) < 1e-7, np.pi**2.0/6.0,
-                         sc.expm1(sc.gammaln(2.0*c+1.0)-2*sc.gammaln(c + 1.0))/c**2.0)
+        gam2k = _lazywhere(
+            abs(c) < 1e-7, (c,), 
+            f=lambda x: np.full_like(x, np.pi**2.0/6.0), 
+            f2=lambda x: sc.expm1(sc.gammaln(2.0*x+1.0)-2*sc.gammaln(x + 1.0))/x**2.0
+        )
         eps = 1e-14
-        gamk = np.where(abs(c) < eps, -_EULER, sc.expm1(sc.gammaln(c + 1))/c)
+        gamk = _lazywhere(
+            abs(c) >= eps, 
+            (c,), 
+            f=lambda x: sc.expm1(sc.gammaln(x + 1))/x, 
+            fillvalue=-_EULER
+        )
 
+        # mean
         m = np.where(c < -1.0, np.nan, -gamk)
+        
+        # variance
         v = np.where(c < -0.5, np.nan, g1**2.0*gam2k)
 
         # skewness
-        sk1 = _lazywhere(c >= -1./3,
-                         (c, g1, g2, g3, g2mg12),
-                         lambda c, g1, g2, g3, g2mg12:
-                             np.sign(c)*(-g3 + (g2 + 2*g2mg12)*g1)/g2mg12**1.5,
-                         fillvalue=np.nan)
-        sk = np.where(abs(c) <= eps**0.29, 12*np.sqrt(6)*_ZETA3/np.pi**3, sk1)
+        def sk1_eval(x):
+            return _lazywhere(
+                x < -1./3, (x,), 
+                f=lambda y: np.full_like(y, np.nan), 
+                f2=lambda y: np.sign(y)*(-g3 + (g2 + 2*g2mg12)*g1)/g2mg12**1.5
+            )
```

Comment:
Something like this?
```suggestion
        def sk1_eval(x):
            return _lazywhere(
                x < -1./3, (x, g1, g2, g3, g2mg12), 
                f=lambda x, g1, g2, g3, g2mg12: np.full_like(x, np.nan), 
                f2=lambda x, g1, g2, g3, g2mg12: np.sign(x) *
                (-g3 + (g2 + 2*g2mg12)*g1)/g2mg12**1.5
            )
```